--- src/ui.cpp
+++ src/ui.cpp
@@ -341,6 +341,13 @@
                 u8g2.setCursor(0, 31);
                 u8g2.print("VOL: ");
                 u8g2.print(localState.params.masterVol);
+                
+                // Print CAN RX_Message
+                u8g2.setFont(u8g2_font_5x7_tr);
+                u8g2.setCursor(66, 30);
+                u8g2.print((char) localState.RX_Message[0]);
+                u8g2.print(localState.RX_Message[1]);
+                u8g2.print(localState.RX_Message[2]);
 
                 // 2. Right Side: Waveform Graph
                 const uint8_t graphStartX = 45;
